ci.nix

```diff
@@ -19,26 +19,28 @@ in
 dimension "System" (systems genericPkgs) (systemName: system:
   let
     packageSet = import ./default.nix { inherit system rev; checkMaterialization = true; };
-    pkgs = packageSet.pkgs;
+    packages = import ./nix { inherit system rev; checkMaterialization = true; };
+    pkgs = packages.pkgs;
+    pkgsLocal = packages.pkgsLocal;
     lib = pkgs.lib;
     platformFilter = platformFilterGeneric pkgs system;
   in
   filterAttrsOnlyRecursive (_: v: platformFilter v) {
-    inherit (packageSet) docs papers dev tests plutus-playground marlowe-playground plutus-scb marlowe-symbolic-lambda;
-    inherit (packageSet.haskell.project) roots;
+    inherit (packageSet) docs papers tests plutus-playground marlowe-playground plutus-scb marlowe-symbolic-lambda;
+    inherit (pkgsLocal.haskell.project) roots;
     # build the shell expression to be sure it works on all platforms
-    shell = import ./shell.nix { inherit packageSet; };
+    shell = import ./shell.nix { };
```

Comment:
Instance 3 of importing "almost everything", since it lets the shell re-import everything instead of passing the already imported stuff in.